### ISSUE TITLE
Validation set for the datasets

### ISSUE THREAD

**Issue by oezyurty** [emadeldeen24/AdaTime#2]
First of all, Thanks a lot for all your effort! 

I just have a small concern regarding the usage of preprocessed versions of the data that you provide. Preprocessed datasets always include train and test splits, however, preprocessing scripts seem to include validation sets as well. (In some cases lines for val set are commented out, in some cases they are not.)

To create validation set (i.e. for early-stopping), what would you suggest me to do? Should I, for instance, further split train split into train and val. In that case, is there anything that I should take into account to prevent information leakage across the splits?

Best wishes,  

PS: To make the issue more concrete, you can see in the WISDM preprocessing script that the dataset is first split into train and test, and the train set is further split into train and val. However, val set is not saved.

**Comment by emadeldeen24** [comment]
Thank you @oezyurty for your question. 
In our AdaTime framework, we don't need a validation set for either source or target domains. 

To be more specific, we want to ensure a realistic evaluation for the unsupervised domain adaptation (UDA) problem. In UDA, we assume that we don't have access to target domain labels, and hence, we can't make use of its validation set to do early stopping.
Instead, we provide other risks to choose the hyperparameters based on them. 
1- Source risk only requires a source test set.
2- DEV risk requires source and target training sets + source test set.
3- Few-shot target risk -- (for semi-supervised DA)
Finally, to calculate the upper limit of the model performance (Target risk), it requires a target domain test set.

I hope this answers your question.

**Comment by oezyurty** [comment]
Dear @emadeldeen24 , thank you for your quick reply! 

Actually, I have still some small concerns. It is not about the credibility of your work, but rather the implication of the results. 

Quick note: Here, I will be referring to validation set for early stopping and hyperparameter selection, and test set for reporting the results. (Some papers use these terms other way around.)

Using the same test set for hyperparameter selection and reporting the result can lead to optimistic results. In other words, the user is in the risk of overfitting to the test set itself. That is why, there should be one left-out set, which should **only** be used for reporting the results, i.e., after hyperparameters have been set and the models have been trained. 

I know you are not using target labels of the target domain (that makes your work clearly more realistic compared to many other UDA papers). However (as one example), you are still in the risk of getting optimistic results for source risk, because you may have chosen the hyperparameters according to the same test set already. This would mean that if we have new data from the same subject, all the models could perform worse than what's reported.

That being said, since you compare all the models with same strategy, the comparative results of your framework should remain valid. (except if there is any model that explicitly favors generalizability more than the others, then we may not realize its performance based on the current setting.)

My intention here was to have nice discussion and clarify if there is any misunderstanding :) In my case, I will probably further divide the test set into val and test set. (If I split the training set instead, validation set would have exactly the same summary statistics as training set => another form of information leakage. )

If you have no further comments, you can safely close this issue. Best wishes!

**Comment by emadeldeen24** [comment]
> Using the same test set for hyperparameter selection and reporting the result can lead to optimistic results.

Agree, but this is not what we do here. We use the target test set to report the results, but we never use it for hyperparameter selection (when using source risk and DEV risk).

> This would mean that if we have new data from the same subject, all the models could perform worse than what's reported.

If the new data is from the same distribution used for training, this should not be the case. But this needs to be tested actually.

> My intention here was to have nice discussion and clarify if there is any misunderstanding

It is always good to have such discussions.

**Comment by emadeldeen24** [comment]
@oezyurty if you are using our codes in your ICLR submission, you should at least cite our work (even from an ethical perspective).

**Comment by oezyurty** [comment]
Hi @emadeldeen24, I have addressed your issue in our GitHub repo with a detailed response. Best regards,
